ethportal-api/src/types/content_key/history.rs

```diff
@@ -0,0 +1,376 @@
+use ethereum_types::H256;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use sha2::{Digest as Sha2Digest, Sha256};
+use ssz::{self, Decode, Encode};
+use ssz_derive::{Decode, Encode};
+use std::fmt;
+
+use crate::types::content_key::error::ContentKeyError;
+use crate::types::content_key::overlay::OverlayContentKey;
+use crate::utils::bytes::{hex_decode, hex_encode, hex_encode_compact};
+
+/// SSZ encoded overlay content key as bytes
+pub type RawContentKey = Vec<u8>;
+
+/// A content key in the history overlay network.
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq)]
+#[ssz(enum_behaviour = "union")]
+pub enum HistoryContentKey {
+    /// A block header with accumulator proof.
+    BlockHeaderWithProof(BlockHeaderKey),
+    /// A block body.
+    BlockBody(BlockBodyKey),
+    /// The transaction receipts for a block.
+    BlockReceipts(BlockReceiptsKey),
+    /// An epoch header accumulator.
+    EpochAccumulator(EpochAccumulatorKey),
+}
+
+impl Serialize for HistoryContentKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_hex())
+    }
+}
+
+impl<'de> Deserialize<'de> for HistoryContentKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let data = String::deserialize(deserializer)?.to_lowercase();
+        let first_two = &data[..2];
+
+        if first_two != "0x" {
```

Comment:
I'm guessing this was just copy/pasted, but maybe update this to use `starts_with`